File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let dummy= new ListNode(-1)
    dummy.next = head;
    let prev = dummy
    let curr = head
    while(curr){
        if(curr.val === val){
            prev.next = curr.next;
            curr= curr.next
        }else{
            prev = curr
            curr = curr.next
        }
    }

    return dummy.next
};

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        tmp = ListNode(-1, head)
        prev = tmp
        while prev.next:
            if prev.next.val != val:
                prev = prev.next
            else:
                prev.next = prev.next.next
      
        
        return tmp.next